services/repository/src/main/java/org/sagebionetworks/file/services/FileUploadServiceImpl.java

```diff
@@ -146,13 +147,16 @@ public UploadDaemonStatus getUploadDaemonStatus(Long userId, String daemonId) th
 	@Override
 	public String getPresignedUrlForFileHandle(Long userId, String fileHandleId) throws NotFoundException {
 		UserInfo userInfo = userManager.getUserInfo(userId);
-		return fileHandleManager.getRedirectURLForFileHandle(userInfo, fileHandleId);
+		FileHandleUrlRequest urlRequest = new FileHandleUrlRequest(userInfo, fileHandleId);
+		return fileHandleManager.getRedirectURLForFileHandle(urlRequest);
 	}
 
 	@Override
 	public String getPresignedUrlForFileHandle(Long userId, String fileHandleId, FileHandleAssociateType fileAssociateType, String fileAssociateId) throws NotFoundException {
 		UserInfo userInfo = userManager.getUserInfo(userId);
-		return fileHandleManager.getRedirectURLForFileHandle(userInfo, fileHandleId, fileAssociateType, fileAssociateId);
+		FileHandleUrlRequest urlRequest = new FileHandleUrlRequest(userInfo, fileHandleId)
+				.withAssociation(fileAssociateType, fileAssociateId);
+		return fileHandleManager.getRedirectURLForFileHandle(urlRequest);
 	}
```

Comment:
integration test for case of with and without association in FileHandleManagerImplAutowireTest